frida_tools/tracer.py

```diff
@@ -514,12 +538,16 @@ def _create_resolver_script(self):
 
 
 class Tracer(object):
-    def __init__(self, reactor, repository, profile, log_handler=None):
+    def __init__(self, reactor, repository, profile, userdata, log_handler=None):
         self._reactor = reactor
         self._repository = repository
         self._profile = profile
         self._script = None
         self._log_handler = log_handler
+        
+        self._userdata = userdata
```

Comment:
Could be simplified to:

```suggestion
        self._userdata = userdata if userdata is not None else ""
```

Reply:
Done.